### ISSUE TITLE
Resnet for non standard images

### ISSUE THREAD

**Issue by sakvaua** [fchollet/keras#5154]
Sorry for asking the question here, but stackoverflow seems to be a little passive when it comes to Keras.
I would like to use ResNet50 on a set of Black and White 16 bit uint images. What Resnet expects as an input? Tensors with zero centered [-127;128] values? [-1;1]? Something else?

**Comment by kirk86** [comment]
ResNet50 expects images as 3D tensors. where the default value of image's width is   `default_size=224` and the minimum width is `min_size=197`

**Comment by sakvaua** [comment]
Thanks. 
Well, if you remove top layers you can feed it even larger images. My question is on the single pixel values. Does is matter if I feed it 0 centered pixels in the range of [-1;1], [-127,128] or something like [-1000,1000]?

**Comment by kgrm** [comment]
The images are converted to float32 internally. Their values can be anything you like, but if you use a pre-trained model, its accuracy will obviously decrease if the value range differs from what it was trained on.

**Comment by sakvaua** [comment]
Thanks. I want to use a pre-trained on imagenet model. What range it was trained on?

**Comment by kgrm** [comment]
Probably uint8 rgb images unless specified otherwise.

**Comment by sakvaua** [comment]
So no rescaling and only mean subtraction?

**Comment by bstriner** [comment]
It will hopefully specify wherever you are getting the model from. If not, you might be able to look through the code and figure out what was done. There are a couple versions out there that have different preprocessing.

I wouldn't expect it to do anything useful, since it is trained on one set of color images and you are testing it out on some black and white images. Are you trying to see if resnet generalizes to B&W? What is the goal?

**Comment by monktastic** [comment]
I'm trying to use ResNet50 from applications and haven't figured out what normalization it expects. I've subtracted out the usual means, tried converting to float32, and converting RGB->BGR, but no luck.

I'm fine-tuning it on two categories and it never exceeds chance (50%).

**Comment by monktastic** [comment]
The problem seems to be limited to the TF Keras. When using base Keras everything works as expected.
